### DISCUSSION TITLE
If you were to start learning programming starting today, where would you start?

### DISCUSSION THREAD

**Discussion by LordNex** [blakeblackshear/frigate#4428] [General]
I've been tinkering with programming since I was 9 on a Commodore 64C and have been in IT for a good 25+ years. But I've never focused on Programming. Now I find its more and more something I want to learn and be proficient at. 

So if you had to start from scratch, what resources, languages, websites, books, classes, etc would you utilize and why?

**Comment by NickM-27** [comment]
For me, as someone who thoroughly enjoys programming and sees it as more than just their job, I think finding something interesting to work on is not only a good motivator but also makes it easier to learn. I took a java class in highschool and it was helpful, but I didn't learn or understand nearly as much compared to the summer after when I started working on Android apps for a problem I wanted to solve myself. Being interested in it personally made it a lot easier to learn and a lot easier to persevere through the difficult parts.

If you were interested in programming as a job, I think it is important to learn languages like C, C++, and even ARM assembly. Doing that in college gave me a lot of respect for how simple modern high level languages make things and also how complicated it is under the hood. To be clear I don't enjoy those but it was important to learn. If it's just for hobby then I don't think it is necessary.

My other piece of advise for someone getting started learning programming is pick an ecosystem that is simple. It is exponentially more difficult to learn a programming language while you're also trying to learn a complicated ecosystem. This is why I wouldn't for example recommend learning java / kotlin on Android as Android itself is full of potholes and difficulties. (same for iOS / swift).
